### PR TITLE
Expanded ML config

### LINKED ISSUE #15: Realize work with k- Nearest Neighbors.

### PR DESCRIPTION
**Pull request by Vasar007**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

Please provide enough information so that others can review your pull request:
- Expanded config, it can store more than one ML algorithms parameters;
- Fixed train methods signatures in models;
- Added parameters for all existed ML models (LinearRegression, kNearestNeighborsRegressor, CatBoostRegressor).

Explain the **details** for making this change. What existing problem does the pull request solve?
Now we can store multiple ML parameters without copy-paste from temporary files with parameters. Also added common pattern to work with new NL algorithms, parsers and metrics.

**Closing issues**
Resolved #15
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/ptrdiff%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/33400860%3Fv%3D4%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40%23issuecomment-415042848%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222018-08-22T14%3A03%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/33400860%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ptrdiff%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%203f6ec3b2332a16c740e741db564212d13a922e0d%20mlalgorithms/shell.py%2031%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40%23discussion_r211964333%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Better%20to%20use%20string%20key.%20not%20a%20int%20value%20in%20index.%20%22%2C%20%22created_at%22%3A%20%222018-08-22T13%3A58%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/33400860%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ptrdiff%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222018-08-22T14%3A02%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/33400860%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ptrdiff%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222018-08-22T14%3A02%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/21951872%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Vasar007%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20mlalgorithms/shell.py%3AL36-61%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40%23issuecomment-415042848%22%2C%20%22https%3A//github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40%23discussion_r211964333%22%2C%20%22https%3A//github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40%23discussion_r211965956%22%2C%20%22https%3A//github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40%23discussion_r211965972%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/ptrdiff'></a>

- [x] <a href='#crh-comment-Pull 3f6ec3b2332a16c740e741db564212d13a922e0d mlalgorithms/shell.py 31'></a> &nbsp;<b><a href='https://github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40#discussion_r211964333'>File: mlalgorithms/shell.py:L36-61</a></b>
- <a href='https://github.com/ptrdiff'></a> Better to use string key. not a int value in index.

<a href='https://www.codereviewhub.com/robot-lab/tinkoff-optimization-of-procurement/pull/40?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/robot-lab/tinkoff-optimization-of-procurement/pull/40?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/robot-lab/tinkoff-optimization-of-procurement/pull/40?approve=0'></a>&nbsp;<a href='https://github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ptrdiff** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/robot-lab/tinkoff-optimization-of-procurement/pull/40'>Refresh</a>.<a href='#crh-update-none'></a>

File: mlalgorithms/shell.py [hunk 1]
```diff
@@ -36,25 +36,25 @@ def __init__(self, existing_model_name=""):
...
-        self._parser_parameters = self._config_parser["parser_parameters"]
+        self._model_parameters = self._config_parser.get_params_for("model")
+        self._parser_parameters = self._config_parser.get_params_for("parser")
 
         if not existing_model_name:
             self._model = self._config_parser.get_instance(
-                self._config_parser["model_name"],
-                self._config_parser["model_module_name"],
-                **self._model_parameters
+                self._model_parameters[0],
```
Comment (ptrdiff) [inline mlalgorithms/shell.py]: Better to use string key. not a int value in index.

Comment (ptrdiff) [inline mlalgorithms/shell.py]: :+1:

Comment (Vasar007) [inline mlalgorithms/shell.py]: :+1:

**Comment by ptrdiff** [comment]
:shipit:

**Pull request merged**